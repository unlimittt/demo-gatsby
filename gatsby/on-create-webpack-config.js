const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
