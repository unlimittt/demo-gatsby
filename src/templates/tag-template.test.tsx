import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import TagTemplate from './tag-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import allMarkdownRemark from '../../jest/__fixtures__/all-markdown-remark';
import pageContext from '../../jest/__fixtures__/page-context';
import type { RenderCallback } from '../types';
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe('TagTemplate', () => {
  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(siteMetadata)
    );
  });

  const props = {
    data: {
      ...allMarkdownRemark
    },
    ...pageContext
  };

  it('renders correctly', () => {
    const wrapper = shallow(<TagTemplate {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
