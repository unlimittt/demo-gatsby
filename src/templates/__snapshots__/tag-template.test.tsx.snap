// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagTemplate renders correctly 1`] = `
<Layout
  description="Test subtitle"
  title="All Posts tagged as \\"test\\" - Page 1"
>
  <Sidebar />
  <Page
    title="test"
  >
    <Feed
      edges={
        Array [
          Object {
            "node": Object {
              "fields": Object {
                "categorySlug": "/test",
                "slug": "/test_0",
              },
              "frontmatter": Object {
                "category": "test",
                "date": "2016-09-01",
                "description": "test_0",
                "title": "test_0",
              },
              "html": "<p>test1</p>",
              "id": "test-1",
            },
          },
          Object {
            "node": Object {
              "fields": Object {
                "categorySlug": "/test",
                "slug": "/test_1",
              },
              "frontmatter": Object {
                "category": "test",
                "date": "2016-09-01",
                "description": "test_1",
                "title": "test_1",
              },
              "html": "<p>test2</p>",
              "id": "test-2",
            },
          },
        ]
      }
    />
    <Pagination
      hasNextPage={true}
      hasPrevPage={true}
      nextPagePath="/page/3"
      prevPagePath="/page/1"
    />
  </Page>
</Layout>
`;
